modules/update-evaluate-availability.adoc

```diff
@@ -54,6 +54,8 @@ Supported but not recommended updates:
   Message: Nodes with substantial numbers of containers and CPU contention may not reconcile machine configuration https://bugzilla.redhat.com/show_bug.cgi?id=2111817#c22
 ----
 
+// to do: determine whether the rest of the lines in this module should still be included, since this is pretty in-depth even for this sort of descriptive doc, according to Trevor.
```

Comment:
This is in reference to [this discussion thread](https://github.com/openshift/openshift-docs/pull/60091#discussion_r1212360392).

Reply:
I'm not sure if doc-tooling allows this, but a possible compromise between my desire to focus on the abstract description and @petr-muller's [desire][1] to give folks a peek under the hood would be to say that everything in the above `oc adm upgrade` output is sourced from ClusterVersion, and link folks over to [the API docs][2] where they can learn about the structure of `status.availableUpdates` on their own.

[1]: https://github.com/openshift/openshift-docs/pull/60091#discussion_r1213221756
[2]: https://docs.openshift.com/container-platform/4.13/rest_api/config_apis/clusterversion-config-openshift-io-v1.html

Reply:
I'm ok with a separate "how does `oc adm upgrade` work?" with the paired examples that demonstrate that currently it's just a pretty-printer for ClusterVersion status.  That is interesting information for folks who think the command is too magical.  I just don't think that delving into that implementation belongs in this *Evaluation of update availability* section.

Reply:
We should discourage folks to run commands like `oc get clusterversion version -o json | jq '.status.availableUpdates'` unless there is no way they can get some information. If the information can retrieved through `oc command` then we do not need to talk about directly querying clusterversion . Because we do not users modifying clusterversion or any underlying resources. If they get in to habit of doing that it is risky for multiple reasons. The UX is hard. They need to parse information which might not be useful for them. They accidentally modify a resource which might cause issues because QE does not test by directly modifying or querying the resources.

Reply:
> We should discourage folks to run commands like ... unless there is no way they can get some information.

It's hard to make people understand how upgrade works without actually showing them the guts of `ClusterVersion`. Context is important. These examples are presented in the context of _"run command to see internals / how things work"_, not _"run command to get the list of available updates"_.

>  I just don't think that delving into that implementation belongs in this Evaluation of update availability section

I agree with this. My synthesis of all these opinions is that we could have a short section on the `ClusterVersion` object itself: its existence, the fact that you should never modify it directly, that CVO operates over it and that `oc adm upgrade` pretty prints it. Then we could clean the other sections from mentioning the resource.